lazy-pages/src/common.rs

```diff
@@ -77,46 +76,100 @@ pub(crate) struct LazyPagesContext {
 }
 
 impl LazyPagesContext {
+    pub fn contexts(
+        &self,
+    ) -> Result<(&LazyPagesRuntimeContext, &LazyPagesExecutionContext), ContextError> {
+        Ok((self.runtime_context()?, self.execution_context()?))
+    }
+
+    pub fn contexts_mut(
+        &mut self,
+    ) -> Result<(&mut LazyPagesRuntimeContext, &mut LazyPagesExecutionContext), ContextError> {
+        let rt_ctx = self
+            .runtime_context
+            .as_mut()
+            .ok_or(ContextError::RuntimeContextIsNotSet)?;
+        let exec_ctx = self
+            .execution_context
+            .as_mut()
+            .ok_or(ContextError::ExecutionContextIsNotSet)?;
+        Ok((rt_ctx, exec_ctx))
+    }
+
+    pub fn runtime_context(&self) -> Result<&LazyPagesRuntimeContext, ContextError> {
+        self.runtime_context
+            .as_ref()
+            .ok_or(ContextError::RuntimeContextIsNotSet)
+    }
+
     pub fn runtime_context_mut(&mut self) -> Result<&mut LazyPagesRuntimeContext, ContextError> {
         self.runtime_context
             .as_mut()
             .ok_or(ContextError::RuntimeContextIsNotSet)
     }
+
     pub fn execution_context(&self) -> Result<&LazyPagesExecutionContext, ContextError> {
         self.execution_context
             .as_ref()
             .ok_or(ContextError::ExecutionContextIsNotSet)
     }
-    pub fn execution_context_mut(
-        &mut self,
-    ) -> Result<&mut LazyPagesExecutionContext, ContextError> {
-        self.execution_context
-            .as_mut()
-            .ok_or(ContextError::ExecutionContextIsNotSet)
-    }
+
     pub fn set_runtime_context(&mut self, ctx: LazyPagesRuntimeContext) {
         self.runtime_context = Some(ctx);
     }
+
     pub fn set_execution_context(&mut self, ctx: LazyPagesExecutionContext) {
         self.execution_context = Some(ctx);
     }
 }
 
 pub(crate) type Weights = [u64; WeightNo::Amount as usize];
-pub(crate) type PageSizes = [NonZeroU32; PageSizeNo::Amount as usize];
 pub(crate) type GlobalNames = Vec<LimitedStr<'static>>;
+pub type PageSizes = [NonZeroU32; PageSizeNo::Amount as usize];
 
 #[derive(Debug)]
 pub(crate) struct LazyPagesRuntimeContext {
     pub page_sizes: PageSizes,
     pub global_names: GlobalNames,
     pub pages_storage_prefix: Vec<u8>,
+    pub program_storage: Box<dyn LazyPagesStorage>,
```

Comment:
This can be done wout box. To be discussed

Reply:
Will be done in next PR